### DISCUSSION TITLE
Stochastic Lanczos for log determinant computation

### DISCUSSION THREAD

**Discussion by rmassidda** [cornellius-gp/gpytorch#1975] [Q&A] [answered]
Hi there! :wave:

I am trying to reproduce the use of stochastic Lanczos over SKI, as in "Scalable Log Determinants for Gaussian Process Kernel Learning" ([Dong et al., 2017](https://proceedings.neurips.cc/paper/2017/file/976abf49974d4686f87192efa0513ae0-Paper.pdf)). I see that GPyTorch provides an implementation in `utils.StochasticLQ`, but it's nowhere referenced in the codebase anymore.

If I correctly understood the workflow, I should extend `InterpolatedLazyTensor` and redefine `inv_quad_logdet` to use stochastic Lanczos. Then, I should derive a new kernel class from `GridInterpolationKernel` that adopts the newly defined lazy tensor.

Does it make sense? If not, could someone give me some hints? :pray:

**Comment by wjmaddox** [comment, marked as answer]
It should be used automatically inside of `inv_quad_logdet` (via `_inv_quad_logdet`). See the code at https://github.com/cornellius-gp/gpytorch/blob/306d7332d74b0d337552911d69d97d939991d031/gpytorch/functions/_inv_quad_logdet.py#L139. Although now the procedure isn't quite that of Dong et al, but rather a variance reduced method introduced in https://arxiv.org/abs/2107.00243.

**Reply by rmassidda** [reply]
Thanks, it's clear now. :)
